### DISCUSSION TITLE
How do I make it so that an action gets triggered for ALL transitions of the same type?

### DISCUSSION THREAD

**Discussion by InfiniteRain** [statelyai/xstate#3653] [Q&A]
Let's say I have a transition `TRANSITION` that is used in two events. I want to make an action called `myAction` that will be triggered whenever `TRANSITION` happens. The only way it's currently possible to do this is by explicitly adding `actions: ["myAction"]` to each event. Is there a way to do it so that I wouldn't need to explicitly write `action: ["myAction"]` in every single event that uses that transition and instead define it once?

**Comment by davidkpiano** [comment]
There currently isn't a way besides what you've described; what would an ideal API look like to you for this?
